dialer/dialer.go

```diff
@@ -11,28 +11,37 @@ import (
 	"io"
 	"net"
 	"runtime/debug"
+	"sync/atomic"
 	"time"
 
 	"github.com/getlantern/golog"
 )
 
 var log = golog.LoggerFor("dialer")
 
+var currentDialer atomic.Value
+
 // New creates a new dialer that first tries to connect as quickly as possilbe while also
 // optimizing for the fastest dialer.
 func New(opts *Options) Dialer {
-	return NewTwoPhaseDialer(opts, func(opts *Options, existing Dialer) Dialer {
+	if currentDialer.Load() != nil {
```

Comment:
Are we assuming that there will only ever be one place `New` is called? (ignoring tests)

Reply:
It's only called when we get new configs